### DISCUSSION TITLE
The value of stateProbs() and getTrProbs() functions can't match with each other?

### DISCUSSION THREAD

**Discussion by xjtusjtu** [bmcclintock/momentuHMM#183] [Q&A] [answered]
Hi Brett,

```r
fitHaggis <- fitHMM(data = processedHaggis, nbStates = 2,
                    dist = list(step = "gamma", angle = "vm"),
                    Par0 = list(step = stepPar0, angle = anglePar0),
                    formula = ~ slope + I(slope^2),
                    estAngleMean = list(angle = TRUE))
 ```
 
Based on the code provided in "Guide to using momentuHMM.pdf - 2.2 HMM specification and fitting," I obtained the fitted model `fitHaggis`. Subsequently, I executed the `stateProbs()` and `getTrProbs()` functions. The results include the probability of the process being in different states at each time point and the transition probability matrix for each time step, as follows: 

```bash
> sp <- stateProbs(fitHaggis)
> trProbs <- getTrProbs(fitHaggis)
> sp[1:5,]
          state 1      state 2
[1,] 7.191524e-01 2.808476e-01
[2,] 2.174842e-06 9.999978e-01
[3,] 1.141769e-01 8.858231e-01
[4,] 8.933096e-01 1.066904e-01
[5,] 1.000000e+00 4.473818e-08
> trProbs[,,1:5]
, , 1

           state 1   state 2
state 1 0.76818444 0.2318156
state 2 0.05158681 0.9484132

, , 2

           state 1   state 2
state 1 0.61340077 0.3865992
state 2 0.03605583 0.9639442

, , 3

           state 1   state 2
state 1 0.60749200 0.3925080
state 2 0.03961385 0.9603862

, , 4

          state 1   state 2
state 1 0.7311696 0.2688304
state 2 0.1072474 0.8927526

, , 5

          state 1   state 2
state 1 0.8384870 0.1615130
state 2 0.2341643 0.7658357

> sp[1,] %*% trProbs[,,1]
       state 1   state 2
[1,] 0.5669297 0.4330703
> sp[1,] %*% trProbs[,,2]
       state 1   state 2
[1,] 0.4512548 0.5487452
```

What confuses me is that `sp[1,] %*% trProbs[,,1]` is not equal to `sp[2,]`. Even if my indexing is incorrect, `sp[1,] %*% trProbs[,,2]` is also not equal to `sp[2,]`. 

If you know the reason for this, could you please enlighten me? Thank you very much!

**Comment by xjtusjtu** [comment]
The same issue arises where `sp[3,] %*% trProbs[,,3] is not equal to sp[4,]` and `sp[3,] %*% trProbs[,,4] is not equal to sp[4,]`. If there's any insight into why this is happening, I would greatly appreciate your explanation.

**Comment by xjtusjtu** [comment]
If I replace `fitHaggis` with `example$m`, the issue still persists.

```R
m <- example$m
sp <- stateProbs(m)
trProbs <- getTrProbs(m)
sp[1:5,]
trProbs[,,1:5]
sp[1,] %*% trProbs[,,1]
sp[1,] %*% trProbs[,,2]
```

**Comment by bmcclintock** [comment, marked as answer]
Unless I'm missing something, the simple answer is that they shouldn't be equal. You should be able to verify this by examining the definitions/derivations of the state probabilities (i.e. "local" decoding based on the forward-backward algorithm) and the state transition probabilities in [Zucchini et al. 2016](https://doi.org/10.1201/b20790) (or see this [tutorial](https://onlinelibrary.wiley.com/action/downloadSupplement?doi=10.1111%2Fele.13610&file=ele13610-sup-0002-Supinfo.pdf) for abbreviated versions).

**Reply by xjtusjtu** [reply]
Thank you very much, brett. I'll read the material carefully!
